torchtune/utils/_checkpointing/_checkpointer.py

```diff
@@ -482,12 +484,57 @@ def save_checkpoint(
                 f"{os.path.getsize(output_path) / 1000**3:.2f} GB "
                 f"saved to {output_path}"
             )
+            # Phi-3-mini uses fused QKV in PEFT, this will not work as expected
+            # if only a proper subset of Q, K, V have been fine-tuned
+            if self._model_type == ModelType.PHI3_MINI:
```

Comment:
Definitely getting very wieldy. As we discussed, probably there is opportunity to refactor checkpointers with a focus on adapters